pkg/recipes/terraform/config/providers/azure.go

```diff
@@ -119,30 +122,67 @@ func fetchAzureCredentials(ctx context.Context, azureCredentialsProvider credent
 			return nil, nil
 		}
 
 		return nil, err
 	}
 
-	if credentials == nil || credentials.ClientID == "" || credentials.TenantID == "" || credentials.ClientSecret == "" {
-		logger.Info("Azure credentials are not registered, skipping credentials configuration.")
+	switch credentials.Kind {
+	case ucp_datamodel.AzureServicePrincipalCredentialKind:
+		if credentials.ServicePrincipal == nil ||
+			credentials.ServicePrincipal.ClientID == "" ||
+			credentials.ServicePrincipal.TenantID == "" ||
+			credentials.ServicePrincipal.ClientSecret == "" {
+			logger.Info("Azure service principal credentials are not registered, skipping credentials configuration.")
+			return nil, nil
+		}
+
+		return credentials, nil
+	case ucp_datamodel.AzureWorkloadIdentityCredentialKind:
+		if credentials.WorkloadIdentity == nil ||
+			credentials.WorkloadIdentity.ClientID == "" ||
+			credentials.WorkloadIdentity.TenantID == "" {
+			logger.Info("Azure workload identity credentials are not registered, skipping credentials configuration.")
+			return nil, nil
+		}
+
+		return credentials, nil
+	default:
+		logger.Info("Azure credential is not supported, skipping credentials configuration, kind: %s", credentials.Kind)
 		return nil, nil
 	}
 
-	return credentials, nil
 }
 
 func (p *azureProvider) generateProviderConfigMap(configMap map[string]any, credentials *credentials.AzureCredential, subscriptionID string) map[string]any {
 	if subscriptionID != "" {
 		configMap[azureSubIDParam] = subscriptionID
 	}
 
-	if credentials != nil && credentials.ClientID != "" && credentials.TenantID != "" && credentials.ClientSecret != "" {
-		configMap[azureClientIDParam] = credentials.ClientID
-		configMap[azureClientSecretParam] = credentials.ClientSecret
-		configMap[azureTenantIDParam] = credentials.TenantID
+	switch credentials.Kind {
+	case ucp_datamodel.AzureServicePrincipalCredentialKind:
+		if credentials.ServicePrincipal != nil &&
+			credentials.ServicePrincipal.ClientID != "" &&
+			credentials.ServicePrincipal.TenantID != "" &&
+			credentials.ServicePrincipal.ClientSecret != "" {
+			configMap[azureClientIDParam] = credentials.ServicePrincipal.ClientID
+			configMap[azureClientSecretParam] = credentials.ServicePrincipal.ClientSecret
+			configMap[azureTenantIDParam] = credentials.ServicePrincipal.TenantID
+		}
+	case ucp_datamodel.AzureWorkloadIdentityCredentialKind:
+		if credentials.WorkloadIdentity != nil &&
+			credentials.WorkloadIdentity.ClientID != "" &&
+			credentials.WorkloadIdentity.TenantID != "" {
+			configMap[azureClientIDParam] = credentials.WorkloadIdentity.ClientID
+			configMap[azureTenantIDParam] = credentials.WorkloadIdentity.TenantID
+
+			// Use AKS Workload Identity for Azure provider
+			// https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/aks_workload_identity#configuring-with-environment-variables
+			configMap[azureUseAKSWorkloadIdentityParam] = true
+			configMap[azureUseCLIParam] = false
```

Comment:
Could these be set by the user? Like can we do `configMap[azureUseAKSWorkloadIdentityParam] = credentials.WorkloadIdentity.useAKSWI` or something like that?

Reply:
I think I'm a bit confused - currently this will auto-detect which credentials are registered (WI or SP) and set the correct values in the provider based on that. I'm not sure what benefit adding this option would have, can you explain more?

Reply:
Same. I'm not sure what this has to do with users.

We allow users to set whatever configuration they want for TF providers. We also fill it in for them for AWS and Azure.